### ISSUE TITLE
Error when running `uv sync --no-install-project`: pyproject.toml file not found

### ISSUE THREAD

**Issue by albertferras-vrf** [astral-sh/uv#6573]
I'm trying the new feature in today's `0.3.3` release: `uv sync --no-install-project` but I could not make it work since it keeps failing because `pyproject.toml` file is not found

The full `Dockerfile` example provided in the documentation here https://docs.astral.sh/uv/guides/integration/docker/#intermediate-layers does not work and raises this error too

**Comment by charliermarsh** [comment]
Apologies, does it work if you copy over the `pyproject.toml`? It might be relying on _discovering_ it but not actually reading from it. Will fix.

**Comment by albertferras-vrf** [comment]
Yes, it works after copying `pyproject.toml`

**Comment by charliermarsh** [comment]
Yes it reads it to discover the project name, so it can pick out the right dependency tree in the lockfile.

**Comment by charliermarsh** [comment]
But it does not require the project _contents_ which is the critical piece that would ruin caching.

**Comment by charliermarsh** [comment]
(The docs are incorrect though.)

**Comment by albertferras-vrf** [comment]
What happens if the project has multiple packages (using workspaces), do I need to copy all other pyproject.toml files too?

**Comment by charliermarsh** [comment]
Let me test it before I try to answer confidently.

**Comment by charliermarsh** [comment]
Yes, but I think we can fix that part.

**Comment by charliermarsh** [comment]
I updated the documentation (and added test coverage) for now. We'll follow-up with some other changes.

**Comment by albertferras-vrf** [comment]
thank you @charliermarsh !

One suggestion: Maybe we can consider storing the root (?) project name in uv.lock so that pyproject.toml is not needed. I have no idea how hard it would be to change the code, but it would make it slightly easier to have a working Dockerfile

**Comment by charliermarsh** [comment]
I think we probably can. Just trying to decide if we _should_. It creates some weirdness if we don't use the `pyproject.toml` to discover the root directory, etc.

**Comment by jaklan** [comment]
@charliermarsh I don't know low-level differences between `uv` and `pdm`, just to mention the latter is able to update venv without `pyproject.toml` with the below command:
> pdm sync --no-self

Btw, what is a difference between these two flags?
```
      --locked                                   Assert that the `uv.lock` will remain unchanged
      --frozen                                   Sync without updating the `uv.lock` file
```

**Comment by jaklan** [comment]
@zanieb thanks, I haven't noticed docs contain extended version of options' help messages

**Comment by zanieb** [comment]
The end of `uv sync --help` suggests the longer option and we cover this at https://docs.astral.sh/uv/getting-started/help/ — let me know if there's anything more we can do to emphasize that it's available though!

**Comment by jaklan** [comment]
@zanieb ach lol, haven't noticed the footer at all 😄 Maybe you can make it bold:

> Use **`uv help sync`** for more details.

or put at the top:

> ❯ uv sync --help
> Update the project's environment
>
> Note: Use **`uv help sync`** for more detailed help.
>
> Usage: uv sync [OPTIONS]
> ...

but generally - it was simply my bad, I was just quickly testing `uv` and haven't got familiar with the CLI and docs yet

**Comment by charliermarsh** [comment]
The biggest hangup to doing this is that you lose any configuration specified in the `pyproject.toml`, which feels like a footgun.
